hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -705,13 +708,28 @@ public synchronized InitResponse init() throws CertificateException {
     if (certificate != null) {
       initCase = initCase | 1;
     }
+
+    Calendar shouldRenewAfter = Calendar.getInstance();
+    shouldRenewAfter
+        .add(Calendar.DAY_OF_YEAR, securityConfig.getRenewalGraceDays());
+    if (initCase == InitCase.ALL.ordinal() &&
+        handleExpiration() && certificate != null &&
+        certificate.getNotAfter().before(shouldRenewAfter.getTime())) {
+      InitCase init = InitCase.EXPIRES;
+      return handleCase(init);
+    }
+
     getLogger().info("Certificate client init case: {}", initCase);
     Preconditions.checkArgument(initCase < 8, "Not a " +
         "valid case.");
     InitCase init = InitCase.values()[initCase];
     return handleCase(init);
   }
 
+  protected boolean handleExpiration() {
```

Comment:
Probably we can remove this API. Return cert expiration state from init(), and let handleCase handle the cert expiration. SCMCertificateClient and DefaultCertificateClient has its own  handleCase implementation.

Reply:
yes, you are right this can be simplified, done.